### ISSUE TITLE
[feature request] frame interpolation

### ISSUE THREAD

**Issue by bajolzas** [byuu/bsnes#102]
I'd like to ask if it is possible to add some kind of frame interpolation, to smooth frame transition.
It would cause some delay, but in turn based games I think it would be worth it, as an option.

**Comment by byuu** [comment]
Why would it be a good idea for turn-based games?

**Comment by bajolzas** [comment]
Well turn based games dont suffer as much from frame delay since they dont need quick input.
Even then I think it would only delay it by one frame (instead of rendering a given frame and outputing it right away, it would instead ouput the middle frame inperpolated using the last and the previous frame; and only then output said frame.)
It would probably work like a shader, but instead of working in space it would work in time.

Are there no other ways to smooth out frame transition? I really thought this was a nice idea :(

**Comment by Screwtapello** [comment]
To output an additional, blended frame between each real frame, bsnes would have to run at 50% speed, or would require a 120Hz monitor. Some TVs do this, but I think they also interpolate *movement* — just crossfading from one frame to the next wouldn't look nearly as good.

Alternatively, bsnes could blend each frame with the previous frame before displaying it, as higan does when emulating LCD motion blur for the Game Boy and Game Boy Advance. However, LCD motion blur is usually regarded as a *bad* thing, and higan only emulates it because there's some GB games that try to use it for a transparency effect. No SNES games are designed for LCD motion blur.

Basically, it's *possible*, but it seems like it would only make the emulation experience worse, so there's not much point in adding it.

**Comment by bajolzas** [comment]
If I'm not mistaken some games ouptut at 60fps but they actually render at 30fps, repeating every rendered frame (at least for me they dont seem to be running at 60fps, could allways be wrong).

My idea would be to replace the first of these optput frames (if my theory is correct) with an interpolated one.

When you use a shader like xbr, you are "guessing" new pixels considering the existing ones. In this case you would "guess" the pixels not considering the pixels existing in the same frame, but those on the previous and following frames.

you can see these examples of what I'm talking about:
https://www.youtube.com/watch?v=h9B9ZIB_Tyc
https://www.youtube.com/watch?v=ZF8VsINBkwM

Sure these are not real time, but there may be some faster variant that could be used.

I guess it would be, at least, worth giving a try; I'll try asking DerKoun about it once he returns.

**Comment by Screwtapello** [comment]
Consoles built for 2D graphics (like the SNES) are very different from consoles built for 3D graphics (like the N64 and Gamecube). Where games for 3D consoles sometimes run at a lower framerate (30fps, 20fps or even 15fps) to achieve prettier graphics, SNES graphics are as pretty as they'll ever be at 60fps, so there's no point in consistently running slower.

There are *some* games in the SNES catalogue that run below 60fps, but not many, and usually only in very busy parts of some levels. bsnes already provides an overclocking option to keep those games running at native 60fps all the time, so interpolation isn't needed.
